core/chains/evm/gas/models.go

```diff
@@ -77,14 +76,15 @@ func NewEstimator(lggr logger.Logger, ethClient feeEstimatorClient, chaintype ch
 		"priceMin", geCfg.PriceMin(),
 		"estimateLimit", geCfg.EstimateLimit(),
 		"daOracleAddress", geCfg.DAOracle().OracleAddress(),
+		"daOracleL1ChainID", geCfg.DAOracle().L1ChainID(),
 	)
 	df := geCfg.EIP1559DynamicFees()
 
 	// create l1Oracle only if it is supported for the chain
 	var l1Oracle rollups.L1Oracle
 	if rollups.IsRollupWithL1Support(chaintype) {
 		var err error
-		l1Oracle, err = rollups.NewL1GasOracle(lggr, ethClient, chaintype, geCfg.DAOracle())
+		l1Oracle, err = rollups.NewL1GasOracle(lggr, ethClient, chaintype, geCfg.DAOracle(), clientsByChainID)
```

Comment:
I feel this and the `IsRollupWithL1Support` do the same thing. We either need to keep  `IsRollupWithL1Support` or preferably the new way we check for DAOracles, otherwise, this logic combined with the added fallbacks will result in an error.

Reply:
make sense, I am thinking about using "0" as fallback value indicate the L1Rollup is not enabled, and have the check inside IsRollupWithL1Support

Reply:
actually, it has to be check in a different function for backward compatibility